main/src/main/scala/sbt/internal/ClassLoaders.scala

```diff
@@ -146,8 +146,9 @@ private[sbt] object ClassLoaders {
         val allTestDependencies = if (layerTestDependencies) allDependenciesSet else Set.empty[File]
         val allRuntimeDependencies = (if (layerDependencies) rawRuntimeDependencies else Nil).toSet
 
+        val scalaLibrarySet = Set(si.libraryJar)
         val scalaLibraryLayer =
-          globalCache.get((List(si.libraryJar), interfaceLoader, resources, tmp))
+          globalCache.get((scalaLibrarySet.toList, interfaceLoader, resources, tmp))
```

Comment:
The one bad thing about this is that if the scala library version is the same as the sbt build scala version, then we somewhat needlessly create a new classloader for the scala library. This could possibly be improved by checking if the appConfiguration.scalaProvider.version matches the scalaVersion and using the sbt library classloader if so. That can be handled in a different PR, however.